### DISCUSSION TITLE
Proposal: Previous consumption/cost change in behaviour

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#984] [Announcements]
**Current Behaviour**

Currently the sensor retrieves data that is x days behind. By default this is 1 day, as meter data is available to OE up to midnight of the previous day. However this can be configured when setting up the integration.

This gets data supplied by the meter directly and does not include data that might be provided by a home mini or pro.

**Issues**

Some people's meters take longer than 24 hours to become available to OE, either temporarily or permanently. This can result in the default behaviour of the sensor to "freeze" as data isn't available in the window that is being searched for. This can be confusing for both new users if their meter is running late and existing users if their meter suddenly reports late.

The configuration of the window to look at is a bit hidden and requires manual intervention.

**Proposed change**

The sensor will retrieve the data for the latest available full day. This means if data suddenly starts running late, the sensor will automatically adapt and still update with the latest full days worth of data that is available. This will result in less confusion when comparing the data with the app or OE website.

If partial days worth of data is available, then the partial data will be ignored. For instance if it's 28/8/24 and data is available up to 27/8/24 23:00, then the sensor will report data 26/8/24 00:00 to 27/8/24 00:00. This scenario is highly unlikely to happen based on how data is reported by your meter to OE, but the scenario should be covered.

**Comment by jwt99412** [comment]
Hopefully this approach will make the data retrieval a bit more stable

**Comment by trevorwarwick** [comment]
Sounds like a good idea...
